### DISCUSSION TITLE
Missing rows in database in query from Next.js

### DISCUSSION THREAD

**Discussion by ryanccn** [supabase/supabase#7853] [Questions] [answered]
Hello,

I've been having problems recently with https://github.com/ryanccn/moddermore, which uses Supabase and Next.js. Sometimes queries for a valid row with an `eq` filter for `id` returns no rows, which results in a 404 in my application. However, when I check the Supabase dashboard, the row *is* there. This is a bit perplexing, and this happens both in production (hosted on Vercel) and in local development.

Can anyone help with this issue? Thanks!

**Comment by GaryAustin1** [comment]
It likely means you do not have a signed in user or the signin process is not finished.  Can you check the id for null before the request to see?  If you have RLS on you could get a similar issue even with the users ID if the jwt is expired.

**Reply by ryanccn** [reply]
No, this is a request from server using the service token, so I don't think RLS will pose any issues.

**Reply by GaryAustin1** [reply]
Your having an actual user id is independent of passing RLS since you have a filter for the id.  You need to check the id is valid before the call.
I assume the 404 is coming from elsewhere in the app as SB should just return empty array without error.

Edit:  Do you have a trace showing a network request with a valid id, service key bearer and getting empty array back?

**Comment by ryanccn** [comment]
This is getting very troublesome lately as it is affecting production heavily.

**Reply by GaryAustin1** [reply]
Have you contacted support?

**Reply by ryanccn** [reply]
Yes. I seem to have fixed the issue after I stopped using `throwOnError`, but I don't see how that would affect this issue

**Comment by ryanccn** [comment]
Would there possibly be a race condition when I create the row with an API response and immediately navigate to that page?

**Reply by GaryAustin1** [reply]
If you don't wait for a response back from the insert there could be.  If you have received the response, then the database has finished its transaction and a read can be done immediately.  It would actually be finished shortly before your client receives the api response back.

**Reply by ryanccn** [reply]
Everything is `await`ed, so I think it does wait for the response... 🤔

**Reply by GaryAustin1** [reply]
It sure might help if you had the network request and response with headers.  I'm guessing you can't do that as this is on a server.
Is the situation an empty array comes back or an error comes back? I've lost track.   
Assuming it is the id in the filter is not found.  Is there anything special about these ids?  How are they generated?  When you insert a new row how do you  know the id to read in your next page?

With it failing on two instances and if without error, sure seems like something on your side.

**Reply by GaryAustin1** [reply]
You might be able to filter the database/api log and see your insert and select request on the database side.

**Comment by ryanccn** [comment, marked as answer]
This problem hasn't occurred in a while, although I haven't made any changes...
